crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -1177,33 +1160,35 @@ impl Target {
     }
 
     /// Checks whether an instruction is supported by the Target based on instruction name and qargs.
-    pub fn instruction_supported(&self, operation_name: &str, qargs: Option<&Qargs>) -> bool {
+    pub fn instruction_supported<'a, T>(&self, operation_name: &str, qargs: T) -> bool
+    where
+        T: Into<QargsRef<'a>>,
+    {
         // Handle case where num_qubits is None by checking globally supported operations
-        let qargs: Option<&Qargs> = if self.num_qubits.is_none() {
-            None
+        let qargs: QargsRef = if self.num_qubits.is_none() {
+            QargsRef::Global
         } else {
-            qargs
+            qargs.into()
         };
         if self.gate_map.contains_key(operation_name) {
-            if let Some(_qargs) = qargs {
-                let qarg_set: HashSet<&PhysicalQubit> = _qargs.iter().collect();
+            if let QargsRef::Concrete(qargs_as_vec) = qargs {
+                let qarg_set: HashSet<&PhysicalQubit> = qargs_as_vec.iter().collect();
```

Comment:
This could likely benefit from a switch to `let QargsRef::Concrete(qargs) = qargs else { return None; };`

Reply:
Addressed in 4006e725879580da3bff7a7c15035216ed7f9bc6